constraints/miden-vm/range-checker.air

```diff
@@ -0,0 +1,31 @@
+mod RangeCheckerAir
+
+fn binary_not(e: scalar) -> scalar:
+    return 1 - e
+
+fn change(e: scalar) -> scalar:
+    return e' - e
+
+ev check_binary(main: [v]):
+    enf v^2 = v
+
+ev transition_8_to_16_bit_validity(main: [t, v]):
+    # Ensure that values in column t can flip from 0 to 1 only once
+    enf t * binary_not(t') = 0
+
+    # Ensure that when column t flips, column v must equal 255
+    enf v = 255 when t' & !t
+
+    # Ensure that when column t flips, v' must be reset to 0
+    enf v' = 0 when t' & !t
+
+ev range_checker(main: [t, s0, s1, v]):
+    # Check selector flags are binary
+    let selectors = [t, s0, s1]
+    enf check_binary([s]) for s in selectors
```

Comment:
can we not just do the following?

```
enf check_binary([s]) for s in [t, s0, s1]
```

Reply:
Currently we don't support inlined vectors as iterables in list or constraint comprehension but I've opened #230.